### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by zarva6596**
[DEMO LINK](https://zarva6596.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,26 +1,93 @@
...
+      });
+    }
+
+    if (name === 'select') {
+      if (value === 'active') {
+        this.setState({ todosNew: todos.filter(todo => !todo.completed) });
+      } else if (value === 'completed') {
+        this.setState({ todosNew: todos.filter(todo => todo.completed) });
+      } else if (value === 'all') {
+        this.setState({ todosNew: todos });
```
Comment (uni-8) [inline src/App.js:56]: switch case

File: src/components/TodoList/TodoList.jsx [hunk 2]
```diff
@@ -1,28 +1,61 @@
...
+            <span className="TodoList__text">{todo.title}</span>
+          </p>
+
+          <button
+            type="button"
+            className="TodoList__button"
+            value={+todo.userId}
+            onClick={userDeteil}
+          >
+            User&nbsp;#1
```
Comment (uni-8) [inline src/components/TodoList/TodoList.jsx:50]: this value should match the id of each user, but now each user has id # 1
image

**Review by uni-8 — CHANGES_REQUESTED** [review]

**Review by uni-8 — APPROVED** [review]

File: src/App.js [hunk 3]
```diff
@@ -1,26 +1,96 @@
...
 import './App.scss';
 import { TodoList } from './components/TodoList';
+import { Search } from './components/Search';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos, getUsers } from './api/api';
 
 class App extends React.Component {
   state = {
+    todosNew: [],
     todos: [],
```
Comment (vkryvytskyy) [inline src/App.js:11]: you should store in state only one todos array

File: src/App.js [hunk 4]
```diff
@@ -1,26 +1,96 @@
...
+
+  changeSelectedUserId = (event) => {
+    this.setState({ selectedUserId: event.target.value });
+  }
+
+  clearUserInfo = () => {
+    this.setState({ selectedUserId: false });
+  }
+
+  hendleFilter = (event) => {
```
Comment (vkryvytskyy) [inline src/App.js:37]:
```suggestion
  handleFilter = (event) => {
```

File: src/App.js [hunk 5]
```diff
@@ -1,26 +1,96 @@
...
+    const { name, value } = event.target;
+    const { todos } = this.state;
+
+    this.setState({ [name]: value });
```
Comment (vkryvytskyy) [inline src/App.js:41]: nice

File: src/components/CurrentUser/CurrentUser.jsx [hunk 6]
```diff
@@ -1,13 +1,30 @@
...
+      {userId}
+    </h2>
 
     <ul>
-      <li>Ervin Howell</li>
-      <li>Shanna@melissa.tv</li>
-      <li>010-692-6593 x09125</li>
+      <li>{users[userId - 1].name}</li>
+      <li>{users[userId - 1].email}</li>
+      <li>{users[userId - 1].phone}</li>
```
Comment (vkryvytskyy) [inline src/components/CurrentUser/CurrentUser.jsx:14]: looks strange. Fetch here only one user, current user I mean. This component should not know about other users

**Review by vkryvytskyy — CHANGES_REQUESTED** [review]
зображення
выбрал юзера с id = 15

File: src/App.js (hunk 4 above)
Comment (zarva6596) [reply src/App.js:37]: done

File: src/components/CurrentUser/CurrentUser.jsx (hunk 6 above)
Comment (zarva6596) [reply src/components/CurrentUser/CurrentUser.jsx:14]: done

File: src/App.js [hunk 7]
```diff
@@ -7,64 +7,47 @@ import { getTodos, getUsers } from './api/api';
 
 class App extends React.Component {
   state = {
-    todosNew: [],
     todos: [],
     users: [],
+    user: {},
```
Comment (vkryvytskyy) [inline src/App.js:12]: use null as default value

File: src/App.js [hunk 8]
```diff
@@ -7,64 +7,47 @@ import { getTodos, getUsers } from './api/api';
...
     selectedUserId: 0,
     input: '',
     select: '',
```
Comment (vkryvytskyy) [inline src/App.js]: store this values in the TodosList components state, entire app shouldn't know about filtering variables

File: src/App.js [hunk 9]
```diff
@@ -7,64 +7,47 @@ import { getTodos, getUsers } from './api/api';
...
   changeSelectedUserId = (event) => {
-    this.setState({ selectedUserId: event.target.value });
+    const { users } = this.state;
+    const { value } = event.target;
+    const user = { ...users.find(us => us.id === +value) };
+
+    this.setState({
+      user,
+      selectedUserId: value,
+    });
```
Comment (vkryvytskyy) [inline src/App.js:34]: consider using API route for fetching specific user

**Review by vkryvytskyy — APPROVED** [review]

**Pull request closed without merging**